tests/src/test/scala/com/nvidia/spark/rapids/RapidsHostMemoryStoreSuite.scala

```diff
@@ -193,6 +193,65 @@ class RapidsHostMemoryStoreSuite extends AnyFunSuite with MockitoSugar {
     }
   }
 
+  test("get memory buffer after host spill") {
+    val sparkTypes = Array[DataType](IntegerType, StringType, DoubleType,
+      DecimalType(ai.rapids.cudf.DType.DECIMAL64_MAX_PRECISION, 5))
+    val spillPriority = -10
+    val hostStoreMaxSize = 1L * 1024 * 1024
+    try {
+      val bm = new RapidsDiskBlockManager(new SparkConf())
+      val (catalog, devStore, hostStore, diskStore) =
+        closeOnExcept(new RapidsDiskStore(bm)) { diskStore =>
+        closeOnExcept(new RapidsDeviceMemoryStore()) { devStore =>
+          closeOnExcept(new RapidsHostMemoryStore(hostStoreMaxSize)) { hostStore =>
+            devStore.setSpillStore(hostStore)
+            hostStore.setSpillStore(diskStore)
+            val catalog = closeOnExcept(
+                new RapidsBufferCatalog(devStore, hostStore)) { catalog => catalog }
+            (catalog, devStore, hostStore, diskStore)
+          }
+        }
+      }
+
+      RapidsBufferCatalog.setDeviceStorage(devStore)
```

Comment:
Do we need to close any previous RapidsBufferCatalog state that may have been leftover from a previous test before clobbering its settings here?

Reply:
Yes, that seems like a good idea. Will do.